beacon_node/beacon_processor/src/lib.rs

```diff
@@ -1246,40 +857,53 @@ impl<T: BeaconChainTypes> BeaconProcessor<T> {
                             if batch_size < 2 {
                                 // One single aggregate is in the queue, process it individually.
                                 if let Some(item) = aggregate_queue.pop() {
-                                    self.spawn_worker(item, toolbox);
+                                    self.spawn_worker(item, idle_tx);
                                 }
                             } else {
                                 // Collect two or more aggregates into a batch, so they can take
                                 // advantage of batch signature verification.
                                 //
                                 // Note: this will convert the `Work::GossipAggregate` item into a
                                 // `Work::GossipAggregateBatch` item.
-                                let mut packages = Vec::with_capacity(batch_size);
+                                let mut aggregates = Vec::with_capacity(batch_size);
```

Comment:
The logic for batch processing attestations and aggregates is largely the same; it's just tweaked a little to handle the `process_individual` and `process_batch` functions.